Library/Homebrew/test/cmd/update-report_spec.rb

```diff
@@ -43,4 +43,42 @@
       expect(new_cache_file).not_to exist
     end
   end
+
+  describe "::migrate_cache_entries_to_symlinks" do
+    let(:formula_name) { "foo" }
+    let(:f) {
+      formula formula_name do
+        url "https://example.com/foo-1.2.3.tar.gz"
+        version "1.2.3"
+      end
+    }
+    let(:old_cache_file) { HOMEBREW_CACHE/"#{formula_name}--1.2.3.tar.gz" }
+    let(:new_cache_file) { HOMEBREW_CACHE/"downloads/5994e3a27baa3f448a001fb071ab1f0bf25c87aebcb254d91a6d0b02f46eef86--foo-1.2.3.tar.gz" }
+    let(:new_cache_symlink) { HOMEBREW_CACHE/"#{formula_name}--1.2.3" }
```

Comment:
Why are there no file extensions now?

Reply:
Why should there be? We are only using the real basename now.

Reply:
It feels strange if we're going to expose these to people to not have them have extensions for ease of extraction etc.

Reply:
Where do we expose this to people?

Reply:
People expect (not unreasonably as it's been the case for all Homebrew's history) for the cache to contain a list of filenames that they can extract.

Reply:
It is harder than it should be when the target of the symlink is prefixed with a long bunch of hex gibberish.  The source tarballs don't need to have the sha256 in their very filenames, surely.

Reply:
@reitermarkus The "guessed" extension is correct the majority of the times. A gentle reminder that formulae account for >95% of installs (comparing with the new cask analytics) and until recently we only used the formulae url's extension for performing downloads anyway. To optimise for the "what if the guessed extension is wrong" case is to reduce the overall aesthetics and usability of these cache files for a minority corner case and that's not a good design. 

Consistency is not king; usability and doing what people expect is.